testsuite/features/step_definitions/common_steps.rb

```diff
@@ -1342,3 +1342,13 @@ def server_secret
     And I wait until the table contains "FINISHED" or "SKIPPED" followed by "FINISHED" in its first rows
   )
 end
+
+Then(/^port "([^"]*)" should be (open|closed)$/) do |port, selection|
+  output, _code = $server.run("ss -l --numeric | grep #{port}  || echo 'closed'", check_errors: false)
+  case selection
+  when 'open'
```

Comment:
It depends :wink: 

```ruby
Then(/^port "([^"]*)" should be open$/) do |port|
  output, _code = $server.run("ss --listening --numeric | grep #{port}  || echo 'closed'", check_errors: false)
  raise "Port '#{port}' not open although it should be!" if output == "closed\n"
end

Then(/^port "([^"]*)" should be closed$/) do |port|
  output, _code = $server.run("ss --listening --numeric | grep #{port}  || echo 'closed'", check_errors: false)
  raise "Port '#{port}' open although it should not be!" if output != "closed\n"
end
```

Reply:
Should I change it or are you ok with how I did it?

Reply:
A matter of personal preference or maybe best practices (if available). I would personally favor a single step in order to avoid duplication and because it seems more elegant, but feel free to ignore me and decide differently.

Reply:
I think the same way as you Johannes. But I guess this is also a matter of personal taste :wink:

Reply:
Third vote for a single step with regex for such kind of positive/negative cases.
It's much easier to maintain and is clear to understand.

I could understand and discuss further if we were in front of a complex step with multiple entries and high cyclomatic complexity, but not for such easy steps :D

Reply:
I really liked the "two steps" wersion a lot more...

Just a remark to close this debate: We currently have no scenario that tests the port being open. So basically, YAGNI... but ok :smile_cat: .